File: services/ai/app/api/v1/health.py
"""
Health check endpoints for AI Service
"""

from fastapi import APIRouter, Depends
from loguru import logger
from app.core.database import get_mongodb, get_redis

router = APIRouter()

@router.get("/")
async def health_check():
    """Basic health check"""
    return {
        "status": "healthy",
        "service": "vision-ai-service",
        "version": "1.0.0"
    }

@router.get("/detailed")
async def detailed_health_check():
    """Detailed health check with database connectivity"""
    health_status = {
        "status": "healthy",
        "service": "vision-ai-service",
        "version": "1.0.0",
        "checks": {
            "mongodb": "unknown",
            "redis": "unknown"
        }
    }
    
    try:
        # Check MongoDB
        db = get_mongodb()
        await db.admin.command('ping')
        health_status["checks"]["mongodb"] = "healthy"
    except Exception as e:
        logger.error(f"MongoDB health check failed: {e}")
        health_status["checks"]["mongodb"] = "unhealthy"
        health_status["status"] = "degraded"
    
    try:
        # Check Redis
        redis_client = get_redis()
        await redis_client.ping()
        health_status["checks"]["redis"] = "healthy"
    except Exception as e:
        logger.error(f"Redis health check failed: {e}")
        health_status["checks"]["redis"] = "unhealthy"
        health_status["status"] = "degraded"
    
    return health_status

@router.get("/ready")
async def readiness_check():
    """Readiness check for Kubernetes"""
    try:
        # Check if all required services are ready
        db = get_mongodb()
        await db.admin.command('ping')
        
        redis_client = get_redis()
        await redis_client.ping()
        
        return {"status": "ready"}
    except Exception as e:
        logger.error(f"Readiness check failed: {e}")
        return {"status": "not_ready", "error": str(e)}

File: backend/src/routes/health.js
const express = require('express');
const router = express.Router();
const logger = require('../utils/logger');

/**
 * @route GET /api/health
 * @desc Get API health status
 * @access Public
 */
router.get('/', (req, res) => {
  try {
    const health = {
      status: 'ok',
      timestamp: new Date().toISOString(),
      service: 'Vision Platform Backend API',
      version: process.env.npm_package_version || '1.0.0',
      environment: process.env.NODE_ENV || 'development',
      uptime: process.uptime(),
      memory: process.memoryUsage(),
      platform: process.platform,
      nodeVersion: process.version
    };

    logger.debug('Health check requested', {
      ip: req.ip,
      userAgent: req.get('User-Agent')
    });

    res.json({
      success: true,
      data: health
    });

  } catch (error) {
    logger.error('Health check failed', { error: error.message });
    
    res.status(500).json({
      success: false,
      error: 'Health check failed'
    });
  }
});

/**
 * @route GET /api/health/detailed
 * @desc Get detailed health status including dependencies
 * @access Public
 */
router.get('/detailed', async (req, res) => {
  try {
    const detailedHealth = {
      status: 'ok',
      timestamp: new Date().toISOString(),
      service: 'Vision Platform Backend API',
      version: process.env.npm_package_version || '1.0.0',
      environment: process.env.NODE_ENV || 'development',
      uptime: process.uptime(),
      memory: process.memoryUsage(),
      platform: process.platform,
      nodeVersion: process.version,
      dependencies: {
        // TODO: Add actual dependency health checks
        mongodb: 'unknown',
        redis: 'unknown',
        minio: 'unknown'
      },
      configuration: {
        corsOrigin: process.env.CORS_ORIGIN ? 'configured' : 'not configured',
        jwtSecret: process.env.JWT_SECRET ? 'configured' : 'not configured',
        microsoftOAuth: process.env.MICROSOFT_CLIENT_ID ? 'configured' : 'not configured',
        analytics: process.env.ENABLE_ANALYTICS === 'true' ? 'enabled' : 'disabled'
      }
    };

    logger.debug('Detailed health check requested', {
      ip: req.ip,
      userAgent: req.get('User-Agent')
    });

    res.json({
      success: true,
      data: detailedHealth
    });

  } catch (error) {
    logger.error('Detailed health check failed', { error: error.message });
    
    res.status(500).json({
      success: false,
      error: 'Detailed health check failed'
    });
  }
});

/**
 * @route GET /api/health/ready
 * @desc Check if service is ready to handle requests
 * @access Public
 */
router.get('/ready', async (req, res) => {
  try {
    // TODO: Add actual readiness checks (database connections, etc.)
    const isReady = true;

    if (isReady) {
      res.json({
        success: true,
        data: {
          status: 'ready',
          timestamp: new Date().toISOString()
        }
      });
    } else {
      res.status(503).json({
        success: false,
        error: 'Service not ready',
        timestamp: new Date().toISOString()
      });
    }

  } catch (error) {
    logger.error('Readiness check failed', { error: error.message });
    
    res.status(503).json({
      success: false,
      error: 'Service not ready',
      timestamp: new Date().toISOString()
    });
  }
});

/**
 * @route GET /api/health/live
 * @desc Check if service is alive (basic liveness probe)
 * @access Public
 */
router.get('/live', (req, res) => {
  res.json({
    success: true,
    data: {
      status: 'alive',
      timestamp: new Date().toISOString()
    }
  });
});

module.exports = router;


